src/compiler/scala/tools/nsc/Global.scala

```diff
@@ -56,6 +56,21 @@ class Global(var currentSettings: Settings, reporter0: LegacyReporter)
     with Reporting
     with Parsing { self =>
 
+  //TODO remove after bootstrapping
+  // sbt uses this compiler to compile its own compiler bridge, therefore we need to use immutable varargs for
+  // compatibility if the library uses them
+  override lazy val isImmutableVarargs: Boolean = try {
```

Comment:
This is quite the hack and it logs a few errors while running the tests but it works well enough to allow bootstrapping.